src/LinuxTracing/LeafFunctionCallManagerTest.cpp

```diff
@@ -175,11 +175,65 @@ TEST_F(LeafFunctionCallManagerTest, PatchCallerOfLeafFunctionReturnsErrorOnSmall
       .Times(1)
       .WillOnce(Return(std::make_optional<bool>(false)));
 
+  // The stack dump is too small, so we are only able to unwind the instruction
+  // pointer.
+  std::vector<unwindstack::FrameData> libunwindstack_callstack;
+  libunwindstack_callstack.push_back(kFrame1);
+
+  // We expect `kStackDumpSize` here as size, as we do not want libunwindstack
+  // to read out of bounds.
+  EXPECT_CALL(unwinder_, Unwind(event_data.pid, &fake_maps, _, _, kStackDumpSize, _, _))
+      .Times(1)
+      .WillOnce(Return(LibunwindstackResult{libunwindstack_callstack,
+                                            unwindstack::ErrorCode::ERROR_INVALID_MAP}));
+
   EXPECT_EQ(Callstack::kStackTopForDwarfUnwindingTooSmall,
             leaf_function_call_manager_.PatchCallerOfLeafFunction(&event_data, &maps_, &unwinder_));
   EXPECT_THAT(event_data.CopyOfIpsAsVector(), ElementsAreArray(callchain));
 }
 
+TEST_F(
+    LeafFunctionCallManagerTest,
+    PatchCallerOfLeafFunctionReturnsSuccessAndPatchesCallchainEvenIfStackDumpDoesContainFullCaller) {
```

Comment:
DoesNot?

Reply:
In that case I would write `DoesNotFullyContainCaller`.

Reply:
Done